Modules/VersionManager/VersionManager.lua

```diff
@@ -0,0 +1,153 @@
+end
+
+function VersionManager:DirectSendVersion(target)
+    LOG:Trace("VersionManager:AnnounceVersion()")
+    local version = CLM.CORE:GetVersion()
+    local message = VersionCommStructure:New(
+        CONSTANTS.VERSIONNING_COMM.TYPE.ANNOUNCE_VERSION,
+        VersionCommAnnounceVersion:New(version.major, version.minor, version.patch, version.changeset))
+    Comms:Send(VERSION_COMM_PREFIX, message, CONSTANTS.COMMS.DISTRIBUTION.WHISPER, target)
+end
+
+function VersionManager:HandleIncomingMessage(message, distribution, sender)
+    LOG:Trace("VersionManager:HandleIncomingMessage()")
+    local mtype = message:Type() or 0
+    if self.handlers[mtype] then
+        self[self.handlers[mtype]](self, message:Data(), sender)
+    end
+end
+
+function VersionManager:OutOfDate(version, disable)
+    LOG:Trace("VersionManager:OutOfDate()")
+    local currentTime = GetServerTime()
+    if disable then
+        if currentTime - self._lastDisplayedMessageD > 300 then
+            LOG:Message("|cffcc0000Your Classic Loot Manager is significantly out of date.|r AddOn communication has been disabled. Version %s is available. Please update as soon as possible.", ColorCodeText(stringifyVersion(version), "00cc00"))
+            self._lastDisplayedMessageD = currentTime
+        end
+        LedgerManager:Cutoff()
+        Comms:Disable()
+    else
+        if currentTime - self._lastDisplayedMessage > 300 then
+            LOG:Message("New version %s of Classic Loot Manager is available. For best experience please update the AddOn.", ColorCodeText(stringifyVersion(version), "00cc00"))
+            self._lastDisplayedMessage = currentTime
+        end
+    end
+end
+
+function VersionManager:HandleAnnounceVersion(data, sender)
+    LOG:Trace("VersionManager:HandleAnnounceVersion()")
+    local currentVersion = CLM.CORE:GetVersion()
+    local receivedVersion = data:Version()
+
+    -- Check if we have older version
+    if currentVersion.major < receivedVersion.major then
+        self:OutOfDate(receivedVersion, true)
+    else -- equal majors
+        if currentVersion.minor < receivedVersion.minor then
+            -- older minor
+            self:OutOfDate(receivedVersion, false)
+        elseif currentVersion.patch < receivedVersion.patch then
+            if receivedVersion.changeset == "" then -- changeset = unofficial or beta. We don't care about that
+                self:OutOfDate(receivedVersion, false)
+            else
+                LOG:Debug("Received version %s-%s", stringifyVersion(receivedVersion), receivedVersion.changeset)
+            end
+        end
+    end
+
+    -- Store received data
+    local profile = ProfileManager:GetProfileByName(sender)
+    if profile then
+        profile:SetVersion(receivedVersion.major, receivedVersion.minor, receivedVersion.patch, receivedVersion.changeset)
+    end
+end
+
+function VersionManager:HandleRequestVersion(data, sender)
```

Comment:
Make this a local function and use an actual function reference in your map of handlers (instead of a string)